src/internal/log/field.go

```diff
@@ -156,3 +157,14 @@ func ContextInfo(ctx context.Context) Field {
 	}
 	return zap.Inline(&contextInfoWrapper{Context: ctx})
 }
+
+// RedactedString returns a field that logs val, redacting all occurrences of redact in val with
+// "[<length> masked bytes]".
+func RedactedString(name, val, redact string) Field {
```

Comment:
I like tacking the redactand at the end, because the signature can be changed to `(string, string, ...string)` in the future if it needs to be, but it can be kept simple for the time being.